# SVM on Handwritten Digits

## Loading libraries

In [1]:
import matplotlib.pyplot as plt
from sklearn.svm import SVC
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn import decomposition
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn import metrics

## Loading data

In [2]:
df = pd.read_csv("E://Telugu Character Recogniton//CSV_dataset//CSV_datasetsix_vowel_dataset_with_class.csv")
df.head()

pixel0  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0     255     255     254     255     255     255     252     255     255   
1     255     255     254     255     255     254     255     255     255   
2     255     254     255     255     255     255     254     255     254   
3     254     255     255     255     254     255     255     255     254   
4     254     255     255     253     255     255     255     254     255   

   pixel9  ...    pixel775  pixel776  pixel777  pixel778  pixel779  pixel780  \
0     255  ...         255       255       255       255       255       255   
1     255  ...         255       255       255       255       255       255   
2     255  ...         255       255       255       255       255       255   
3     255  ...         255       255       255       255       255       255   
4     255  ...         255       255       255       255       255       255   

   pixel781  pixel782  pixel783  class  
0       255       255       255      1  
1       255       255       255      1  
2       255       255       255      1  
3       255       255       255      1  
4       255       255       255      1  

[5 rows x 785 columns]

## Splitting data into 'train' and 'test'

In [3]:
pix=[]
for i in range(784):
    pix.append('pixel'+str(i))
features=pix
X = df.loc[:, features].values
y = df.loc[:,'class'].values

X_train, X_test, y_train, y_test = train_test_split(
 X, y, test_size = 0.25, random_state = 100)
y_train=y_train.ravel()
y_test=y_test.ravel()

## Building classifier

In [4]:

svm_model = SVC(kernel = 'poly', C = 1, gamma=2).fit(X_train, y_train)
svm_predictions = svm_model.predict(X_test)


## Finding accuracy of the model

In [5]:
accuracy = svm_model.score(X_test, y_test)
print('Accuracy: ',accuracy*100)

Accuracy:  89.33333333333333


## <font color='tomato'>Accuracy of the model is: </font><font color='MediumSpringGreen'>89.333333333</font>

## Conclusion:
* <font color='navy' face='lucida console'><strong>Support Vector Machine(SVM) performs well on the small dataset when compared to other models like K-NNC and Neural Neetworks.</strong></font>